Housekeeping

erooke opened the pull request:
While getting to grips with the codebase I noticed some small fixes which could
be made. Instead of opening a PR for each I thought I would just open one PR
addressing all of them. If any changes aren't desired I can remove them from
the request or open a bunch of smaller PRs if that's preferable.

## Changes

- remove bare exceptions
- explicitly state `stacklevel=2` in warnings
- remove function calls from default arguments
- remove unused imports
- make consistent pattern for `not in` and `is not None`
- remove unreachable code


## Rational

Replaced all bare exceptions with `except Exception` most exceptions
derive from this[^1], the ones which don't are not intended to be caught.
These are things Like `SystemExit` and `KeyboardInterrupt`. Catching these
makes it hard to kill the python program.

The default behavior for `warnings.warn` is to only print out the location
where the warning was issued. This is usually not the most helpful behavior as
it does not tell the consumer of `build123d` what they did to trigger the
warning. By setting `stacklevel=2` the call site will be marked with the warning
which is almost surely what is desired[^2].

Having a function call as a default argument causes the function to be called
once (upon declaration) which leads to confusing behavior. Instead we pass in
`None` and call the default function in the function body. This also helps
prevent accidentally mutating the default argument.

[^1]: https://docs.python.org/3/library/exceptions.html#exception-hierarchy
[^2]: https://docs.python.org/3/library/warnings.html#available-functions

erooke committed:
fix: remove bare exceptions

Replaced all bare exceptions with `except Exception` most exceptions
derive from this[^1], the ones which don't are not intended to be caught.
These are things Like `SystemExit` and `KeyboardInterrupt`

[^1]: https://docs.python.org/3/library/exceptions.html#exception-hierarchy

erooke committed:
fix: explicitly state stacklevel=2 in warnings

The default behavior for `warnings.warn` is to only print out the location
where the warning was issued. This is usually not the most helpful behavior as
it does not tell the consumer of build123d what they did to trigger the
warning. By setting stacklevel=2 the call site will be marked with the warning
which is almost surely what is desired.

erooke committed:
fix: remove unused imports

erooke committed:
nit: consistent  pattern

`not x is None` -> `x is not None` and `not x in set` -> `x not in set`

erooke committed:
fix: remove unreachable code

erooke committed:
fix: avoid date.today() as default arg

Having a function call as a default argument causes the function to be called
once (upon declaration) which leads to confusing behavior. Instead we pass in
`None` and call `date.today()` in the function body.

jdegenstein commented:
Thanks for this PR. At first glance many of these changes look good, although there are some missing lines for test coverage (PRs should typically have 100% test coverage).

> - remove bare exceptions

These look generally good to me
> - explicitly state stacklevel=2 in warnings

I also think this change is great
> - remove function calls from default arguments

This makes sense too
> - remove unused imports

I will wait for @gumyr to comment on this, as there are a few cases where we might not want to eliminate the imports.
> - make consistent pattern for not in and is not None

This looks good to me.
> - remove unreachable code

Did not review this one specifically yet. Will wait for @gumyr on this one too.

erooke commented:
> PRs should typically have 100% test coverage

Sounds good, I'll work on fixing the test coverage.

gumyr commented on src/build123d/joints.py:
```diff
@@ -677,8 +680,8 @@ def symbol(self) -> Compound:
         label: str,
-        to_part: Union[Solid, Compound] = None,
-        joint_location: Location = Location(),
+        to_part: Union[Solid, Compound, None] = None,
```
Although I'm not a big fan of Optional as it seems redundant, I believe this is the correct way to do this:
```py
    to_part: Optional[Union[Solid, Compound]] = None,
    joint_location: Optional[Location] = None,
```

gumyr commented on src/build123d/joints.py:
```diff
@@ -75,8 +75,8 @@ def symbol(self) -> Compound:
         label: str,
-        to_part: Union[Solid, Compound] = None,
-        joint_location: Location = Location(),
+        to_part: Union[Solid, Compound, None] = None,
```
Optional?

gumyr reviewed (changes requested):
Thank you! There are lots of nice little improvements here.

erooke committed:
fix: remove location() call from default type

Having a function call as a default argument causes the function to be called
once (upon declaration) which leads to confusing behavior. Instead we pass in
`None` and call `Location` in the function body. This avoids potential issues
with mutating `location` as well.

erooke commented:
Addressed the code review and fixed the missing coverage on the drawing default date behavior. The rest of the missing coverage has to do with warnings which are raised due to OCP behavior (invalid meshes and the like). These paths had no coverage pre-pr and its going to take me a little longer to figure out how to test those particular branches.

gumyr commented:
Thanks for improving the code. These changes make things just a little better in many ways!
